bundles/binding/org.openhab.binding.panstamp/src/main/java/org/openhab/binding/panstamp/internal/PanStampBinding.java

```diff
@@ -0,0 +1,566 @@
+		startDebug(cfg);
+		if (network != null) {
+			for (PanStampBindingProvider prov : providers) {
+				for (String itemName : prov.getItemNames()) {
+					PanStampBindingConfig<?> conf = prov.getConfig(itemName);
+					if (!network.hasDevice(conf.getAddress())) {
+						try {
+							network.addDevice(new PanStamp(network, conf
+									.getAddress()));
+							logger.debug("Added device {} to network",
+									conf.getAddress());
+						} catch (NetworkException e) {
+							logger.error(e.getMessage());
+						}
+					}
+				}
+			}
+		}
+
+	}
+
+	private void stop() {
+		stopDebug();
+		stopNetwork();
+	}
+
+	private void startNetwork(final PanStampBindingSettings cfg) {
+		try {
+			logger.debug("startNetwork()");
+			modem = new SerialModem(cfg.serialPort, cfg.serialSpeed);
+			modem.open();
+			logger.info("Opened panStamp network on serial port {} at {}bps",
+					cfg.serialPort, cfg.serialSpeed);
+			setupModem(cfg);
+			network = Network.create(modem);
+			setupDirectories(cfg);
+			network.setDeviceStore(store);
+			logger.info("Configured binding product codes");
+			network.addListener(networkListener);
+			network.open();
+			logger.info("Opened network");
+//			setProperlyConfigured(true);
+		} catch (NetworkException e) {
+			logger.error("Error connecting to serial network on {}: {}",
+					cfg.serialPort, e.getMessage());
+
+		} catch (SwapException e) {
+			logger.error(
+					"Error reading modem settings for serial network on {}: {}",
+					cfg.serialPort, e.getMessage());
+		} catch (Throwable e) {
+			logger.error("Fatal error: {}", e.getMessage(), e);
+		}
+	}
+
+	private void stopNetwork() {
+		logger.debug("stopNetwork()");
+		try {
+			network.close();
+			network = null;
+			logger.info("Closed panStamp network");
+		} catch (ModemException e) {
+			logger.error("Error closing panStamp network: " + e.getMessage(), e);
+		} finally {
+//			setProperlyConfigured(false);
+		}
+	}
+
+	private void startDebug(final PanStampBindingSettings cfg) {
+		if (cfg.debugEnabled) {
+			try {
+				debugServer = new TcpServer(network.getSWAPModem(),
+						cfg.debugPort);
+				logger.info("TCP Debug enabled on port {}", cfg.debugPort);
+			} catch (TcpException e) {
+				logger.error("Error creating debug service:  {}"
+						+ e.getMessage());
+			}
+		}
+	}
+
+	private void stopDebug() {
+		try {
+			if (debugServer != null) {
+				debugServer.close();
+				debugServer = null;
+			}
+		} catch (TcpException e) {
+			logger.error("Error closing debug port: " + e.getMessage(), e);
+		}
+	}
+
+	private void setupModem(PanStampBindingSettings cfg) {
+		try {
+			ModemSetup setup = modem.getSetup();
+			if (cfg.networkId != -1) {
+				setup.setNetworkID(cfg.networkId);
+			}
+			if (cfg.networkDeviceAddress != -1) {
+				setup.setDeviceAddress(cfg.networkDeviceAddress);
+			}
+			if (cfg.networkChannel != -1) {
+				setup.setChannel(cfg.networkChannel);
+			}
+			if (cfg.networkSecurityOption != -1) {
+				// FIXME
```

Comment:
I've removed the securityOption setting, since the underlying panStamp library does not currently support the functionality. When the library does support it, I'll bring it back.